scripts/foundry/stylus/setCacheManager/README.md

```diff
@@ -0,0 +1,36 @@
+# Adding the wasm CacheManager
+
+This script empowers `AddWasmCacheManagerAction` contract which enables CacheManager for the Orbit chain. Please note that the prerequisite for this action is running the ArbOS 31 version which introduces Stylus support.
+
+## How to use it
+
+1. Setup .env according to the example files, make sure you provide correct values. You can use default values for `INIT_CACHE_SIZE` and `INIT_DECAY`.
+
+`DeployAddWasmCacheManagerAction.s.sol` script deploys `AddWasmCacheManagerAction` contract. It can be executed in this directory like this:
+
+```bash
+forge script --sender $DEPLOYER --rpc-url $CHILD_CHAIN_RPC --broadcast --slow ./DeployAddWasmCacheManagerAction.s.sol -vvv --verify --broadcast
+# use --account XXX / --private-key XXX / --interactive / --ledger to set the account to send the transaction from
+```
+
+This would deploy the action. Update your .env file with the address of the upgrade action.
+
+2. Next step is to execute the ArbOs upgrade action. Make sure your .env file contains `UPGRADE_ACTION_ADDRESS` and `CHILD_UPGRADE_EXECUTOR_ADDRESS` vars. Assumption is that child chain UpgradeExecutor is the arbowner, and there is an EOA which has executor rights on the child chain UpgradeExecutor. Upgrade can be executed using `cast` CLI command (part of Foundry installation), using the owner account (the one with executor rights on child chain UpgradeExecutor) to send the transaction:
```

Comment:
@gvladika I'm a bit confused here - is the general procedure to:
1. Upgrade the Nitro node
2. Upgrade nitro-contracts to 2.1.0 & set the WASM module root
3. Use `DeployAddWasmCacheManagerAction.s.sol` to deploy the `AddWasmCacheManagerAction` contract
4. Schedule the ArbOS upgrade using `scripts/foundry/arbos-upgrades/at-timestamp`

If so, im gonig to make this call out in our docs.arbitrum.io

Reply:
so because we want to ship 2.1.0 (for fast confirmation) before ArbOS31 is fully supported by nitro, there are 2 separate procedure

v2.1.0 Contract Upgrade:
1. Upgrade the Nitro node
1. Upgrade nitro-contracts to 2.1.0 & set the WASM module root

ArbOS31 Upgrade:
1. (prerequisite) v2.1.0 Contract Upgrade
1. Schedule the ArbOS upgrade using scripts/foundry/arbos-upgrades/at-timestamp
1. Use DeployAddWasmCacheManagerAction.s.sol to deploy the AddWasmCacheManagerAction contract
1. Execute AddWasmCacheManagerAction

The specific text you quoted is wrong, it should be:
```suggestion
2. Next step is to execute the AddWasmCacheManagerAction. Make sure your .env file contains `UPGRADE_ACTION_ADDRESS` and `CHILD_UPGRADE_EXECUTOR_ADDRESS` vars. Assumption is that child chain UpgradeExecutor is the arbowner, and there is an EOA which has executor rights on the child chain UpgradeExecutor. Upgrade can be executed using `cast` CLI command (part of Foundry installation), using the owner account (the one with executor rights on child chain UpgradeExecutor) to send the transaction:
```

Reply:
Right, fixed it here https://github.com/OffchainLabs/orbit-actions/pull/16/commits/f4701a56d3442836137fd720542ca39cf301b9e3